flake.nix

```diff
@@ -15,14 +27,183 @@
           nativeBuildInputs = [ pkgs.cmake ];
           buildInputs = [ pkgs.openssl ];
 
-          cmakeFlags = [
-            "-DBUILD_SHARED_LIBS=ON"
-            "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
-            "-DS2N_NO_PQ=1" # TODO: set when system like aarch64/mips,etc
-          ];
+          configurePhase = ''
+            cmake -S . -B./build \
+                  -DBUILD_SHARED_LIBS=ON \
+                  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+                  -DS2N_NO_PQ=1
+          ''; # TODO: set when system like aarch64/mips,etc
+
+          buildPhase = ''
+            cmake --build ./build -j $(nproc)
+          '';
+
+          installPhase = ''
+            cmake --install ./build --prefix $out
+          '';
+
+          checkPhase = ''
+            echo Not running tests here. Run `nix develop` to run tests.
+          '';
 
           propagatedBuildInputs = [ pkgs.openssl ];
+        };
+        devShells.default = pkgs.mkShell {
+          # This is a development enviroment shell which should be able to:
+          #  - build s2n-tls
+          #  - run unit tests
+          #  - run integ tests
+          #  - do common development operations (e.g. lint, debug, and manage repos)
+          inherit system;
+          shellHook = ''
```

Comment:
These functions should be moved to a separate file so you get syntax highlighting and it's possible to use/test them outside of nix. I have a little helper I like to use:

```nix
let
  writeMapFile = map: path: pkgs.writeText (baseNameOf path) (map (builtins.readFile path));
  writeFile = (writeMapFile (v: v));
  writeScriptFile = path: pkgs.writeScript (baseNameOf path) (builtins.readFile path);

....

  shellHook = ''
  source ${writeFile ./path/to/script.sh}
  ''
```